src/main/java/com/vdmytriv/bookstoreapp/repository/BookRepositoryImpl.java

```diff
@@ -35,6 +36,15 @@ public List<Book> findAll() {
         } catch (PersistenceException e) {
             throw new RepositoryException("Couldn't find books ", e);
         }
+    }
 
+    @Override
+    public Optional<Book> findById(Long id) {
+        try {
+            Book book = entityManager.find(Book.class, id);
+            return Optional.ofNullable(book);
+        } catch (PersistenceException e) {
+            throw new RepositoryException("Couldn't find book " + id, e);
```

Comment:
in DAO layer use `DataProcessingException`
```suggestion
            throw new DataProcessingException("Couldn't find book " + id, e);
```